aspnetcore/security/how-to-choose-identity-solution.md

```diff
@@ -0,0 +1,114 @@
+
+For the authentication, the OAuth 2.0 specifification defines two types of clients.
+
+* **Confidential clients**, like server-rendered web apps, are able to keep secrets. The secret is stored on the server and therefore is never shared with the user.
+* Most other clients, including mobile and desktop, are **public clients** that are not suitable for storing secrets because they can be accessed on the user. Simply encrypting the data is not considered strong enough protection. These clients require a different authentication flow that uses tokens.
+
+Tokens are containers with metadata that are explicitly passed through the headers or body of HTTP requests. The main advantage of tokens over cookies is that they are not tied to a specific app or domain. Instead, tokens are usually *signed* with asymmetric cryptography. Asymmetric cryptography uses a combination of a private key known only to the signer, and a public key which everyone can know.
+
+The token can't be tampered with due to the private key. The public key:
+
+* Makes it possible to validate the token to ensure it hasn't been changed.
+* Guarantees that it was generated by the entity holding the private key.
+
+The main disadvantage to using tokens is that they require a token server to both issue and provide validation for tokens. The token server must be installed, configured, and maintained.
+
+A common reason a token server is required is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens. If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
+
+* Can be used by mobile and desktop clients.
+* Isn't suitable for managing access from external apps.
+
+Another reason a token server is required is for sharing sign-ins with other apps. Commonly referred to as *single sign on*, this feature enables users to:
+
+* Sign in once with a web app's form.
+* Use the resulting credentials to authenticate with other apps without having to sign-in again or choose a different password.
+
+A token server is typically preferred to provide a secure and scalable solution for single sign on. 
+
+For apps that don't expose web-based APIs that are consumed by other apps and don't share logins with other apps, the simplest way to quickly secure an app is to use the built-in ASP.NET Core Identity provider. Otherwise, a token server provided by a third-party identity management solution is needed. Token servers are available as:
```

Comment:
```suggestion
For apps that don't share logins with other apps, the simplest way to quickly secure an app is to use the built-in ASP.NET Core Identity provider. Otherwise, a token server provided by a third-party identity management solution is needed. Token servers are available as:
```

Multiple apps should work fine with `MapIdentityApi` as long as they're all accessing the same thing.